aten/src/ATen/native/cpu/AdaptiveMaxPoolKernel.cpp

```diff
@@ -247,33 +252,33 @@ void cpu_adaptive_max_pool_channels_last<BFloat16>(
       int64_t iw0 = start_index(ow, output_width, input_width);
       int64_t iw1 = end_index(ow, output_width, input_width);
 
-      BFloat16* out = output_data + i * channels;
+      scalar_t* out = output_data + i * channels;
       int64_t* ind = indices_data + i * channels;
 
       // Pass I: init out lane
       iVec index0_ivec = iVec(ih0 * input_width + iw0);
-      fVec max_fvec = fVec(-std::numeric_limits<float>::infinity());
+      fVec max_fvec = fVec(-std::numeric_limits<param_t>::infinity());
```

Comment:
This change would have made sense, if you've tried to combine two templates (for full precision and half precision floats) into one, but as is, they are somewhat meaningless

Reply:
Yes, `param_t` is always float.